distribution/docker/build.gradle

```diff
@@ -165,6 +168,30 @@ tasks.register("copyKeystore", Sync) {
   }
 }
 
+tasks.register("ensureLog4jConfigurationSync") {
+  /*
+   * Whenever the x-pack/plugin/core/src/main/config/log4j2.properties is changed it might be that the
+   * distribution/docker/src/docker/config/log4j2.properties file needs changing as well.
+   *
+   * This task fails if the x-pack/plugin/core/src/main/config/log4j2.properties file
+   * has been modified. Please ensure that the distribution/docker/src/docker/config/log4j2.properties
+   * file is also modified such that it reflects the same changes.
+   *
+   * SET THIS to the value of checksumOriginalLog4j after the two files have been confirmed as equivalent
+   */
+  def validatedChecksumOriginalLog4j = "+GJ/QjaApSAiejScjiwFEg=="
+  doLast {
+    def originalLog4j = project(":x-pack:plugin:core").file('src/main/config/log4j2.properties').getBytes()
+    def checksumOriginalLog4j = Base64.getEncoder().encodeToString(MessageDigest.getInstance("MD5").digest(originalLog4j))
+    if (false == validatedChecksumOriginalLog4j.equals(checksumOriginalLog4j)) {
+      throw new GradleException("The src/main/config/log4j2.properties file from the x-pack:plugin:core has been changed. " +
```

Comment:
We should give some indication as to where "this task's declaration" lives. Perhaps provide the relative path to this build.gradle file in the exception message? Or instead externalize the checksum in a file (like we do for dependency artifacts) and just point to that so folks don't actually have to change build scripts.